tests/bgp/test_bgp_4-byte_as_trans.py

```diff
@@ -0,0 +1,290 @@
+
+    dut_ip_bgp_sum = duthost.shell('show ip bgp summary')['stdout']
+    neigh_ip_bgp_sum = nbrhosts[neigh1]["host"].shell('show ip bgp summary')['stdout']
+    neigh2_ip_bgp_sum = nbrhosts[neigh1]["host"].shell('show ip bgp summary')['stdout']
+    with open(bgp_id_textfsm) as template:
+        fsm = textfsm.TextFSM(template)
+        dut_bgp_id = fsm.ParseText(dut_ip_bgp_sum)[0][0]
+        neigh_bgp_id = fsm.ParseText(neigh_ip_bgp_sum)[1][0]
+        neigh2_bgp_id = fsm.ParseText(neigh2_ip_bgp_sum)[1][0]
+
+    dut_ipv4_network = duthost.shell("show run bgp | grep 'ip prefix-list'")['stdout'].split()[6]
+    dut_ipv6_network = duthost.shell("show run bgp | grep 'ipv6 prefix-list'")['stdout'].split()[6]
+    neigh_ipv4_network = nbrhosts[neigh1]["host"].shell("show run bgp | grep 'ip prefix-list'")['stdout'].split()[6]
+    neigh_ipv6_network = nbrhosts[neigh1]["host"].shell("show run bgp | grep 'ipv6 prefix-list'")['stdout'].split()[6]
+
+    setup_info = {
+        'duthost': duthost,
+        'neighhost': neighbors[neigh1],
+        'neigh2host': neighbors[neigh2],
+        'neigh1': neigh1,
+        'neigh2': neigh2,
+        'dut_asn': dut_asn,
+        'neigh_asn': neigh_asn[neigh1],
+        'neigh2_asn': neigh_asn[neigh2],
+        'asn_dict':  neigh_asn,
+        'neighbors': neighbors,
+        'cli_options': cli_options,
+        'dut_ip_v4': dut_ip_v4,
+        'dut_ip_v6': dut_ip_v6,
+        'neigh_ip_v4': neigh_ip_v4,
+        'neigh_ip_v6': neigh_ip_v6,
+        'neigh2_ip_v4': neigh2_ip_v4,
+        'neigh2_ip_v6': neigh2_ip_v6,
+        'peer_group_v4': peer_group_v4,
+        'peer_group_v6': peer_group_v6,
+        'asic_index': asic_index,
+        'dut_bgp_id': dut_bgp_id,
+        'neigh_bgp_id': neigh_bgp_id,
+        'neigh2_bgp_id': neigh2_bgp_id,
+        'dut_ipv4_network': dut_ipv4_network,
+        'dut_ipv6_network': dut_ipv6_network,
+        'neigh_ipv4_network': neigh_ipv4_network,
+        'neigh_ipv6_network': neigh_ipv6_network
+    }
+
+    logger.debug("DUT BGP Config: {}".format(duthost.shell("show run bgp", module_ignore_errors=True)['stdout']))
+    logger.debug("Neighbor BGP Config: {}".format(nbrhosts[neigh1]["host"].shell("show run bgp")['stdout']))
+    logger.debug('Setup_info: {}'.format(setup_info))
+
+    yield setup_info
+
+    # restore config to original state
+    config_reload(duthost)
+    config_reload(neighbors[neigh1], is_dut=False)
+
+    # verify sessions are established
+    bgp_facts = duthost.bgp_facts(instance_id=asic_index)['ansible_facts']
+    for k, v in bgp_facts['bgp_neighbors'].items():
+        if v['description'].lower() not in skip_hosts:
+            logger.debug(v['description'])
+            assert v['state'] == 'established'
+
+
+def test_4_byte_asn_translation(setup):
+    # copy existing BGP config to a new 4-byte ASN.  Use existing route-maps for consistancy.
+    cmd = 'vtysh{} \
```

Comment:
there is a space needed after `vtysh` : cli_options = "-n " + duthost.get_namespace_from_asic_id(asic_index)

Reply:
Yes, sorry I missed this.